Recruit-Training/Advanced-Recruit-Training/Embedded-Teaser/Training/Core/Src/Training.c

```diff
@@ -3,4 +3,65 @@
 void trainingTask(uint8_t* data)
 {
 
+validData = 0b00000001;
+outputArray[0] = 0;
+outputArray[1] = 0;
+outputArray[2] = data[2];
+
+// Motor Velocity and direction, used to check if motors are in sync
+uint8_t velocity1 = data[0] & 0b01111111;
+uint8_t velocity2 = data[1] & 0b01111111;
+
+// Motor On/Off
+uint8_t motor1 = data[0] & 0b10000000;
+uint8_t motor2 = data[1] & 0b10000000;
```

Comment:
You can probably do a bit better with the variable name here. motor1State is probably a bit more descriptive.